### ISSUE TITLE
Release note and backward compatibility

### ISSUE THREAD

**Issue by damienpicard** [open-ideas/IDEAS#429]
As we want to include backward non-compatible changes, I propose that we list in this issues all the major changes and the adaptation needed for make our models compatibles

**Comment by rubenbaetens** [comment]
The `mSenFac` change i proposed in #422 is actually backward compatible.

**Comment by rubenbaetens** [comment]
I propose to make a master 0.4.0 and only make a 1.0.0 once Annex 60 has an official or final release.

**Comment by rubenbaetens** [comment]
Restructuring in #417 included following changes which need to be included in the release notes:

- [ ] Moved `IDEAS.HeatingSystems` to `IDEAS.Circuits`
- [ ] Moved `IDEAS.VentilationSystems` to `IDEAS.Circuits`
- [ ] Moved `IDEAS.Fluid.BaseCircuits` to `IDEAS.Circuits`
- [ ] Moved `IDEAS.Climate` to `IDEAS.BoundaryConditions`
- [ ] Moved `IDEAS.Occupants` to `IDEAS.BoundaryConditions`
- [ ] Replaced `IDEAS.Constants` by `IDEAS.Types`

**Comment by damienpicard** [comment]
The new multi gain feature for multilayer requires the following changes:

- Remove the locGain and nGain from all ConstructionType records which do not have embedded gains
- For all ConstructionType  with embedded gain, make locGain an array (e.g..: locGain = {4})
- for all connector connection the embedded port of the internalWalls, externalWalls, boundaryWalls or slabOnGround, change .port_emb by .port_emb[1].

**Comment by damienpicard** [comment]
Everybody should mention in this issue that his/her models are compatible with the new develop (before making a new release).

**Comment by Mathadon** [comment]
[ ] merged annex60 up to commit 00be78c4f27643d8319999fefb96fb4effea4c79

**Comment by damienpicard** [comment]
changes at constructionType:

1) Add a value to parameter incLastLay for all floor and ceiling constructionType to help you to check the inclination of the surface.

**Comment by rubenbaetens** [comment]
@damienpicard what's the purpose of this parameter `incLastLay` in comparison to `inc` ?

**Comment by Mathadon** [comment]
This parameter verifies that `inc` corresponds to `incLastLay`. This way the user can indicate the intended orientation when creating the structure record and Dymola will check if that record is used in the correct way, according to this definition. This is especially important when connecting floors/ceilings.

**Comment by rubenbaetens** [comment]
@Mathadon Is it correct that:

- [ ] `Fluid.Interfaces.FourPort` has changed name into `Fluid.Interfaces.PartialFourPort` ?

All basecircuits extend from this, but as there are no unittests yet, this didn't show up.

**Comment by Mathadon** [comment]
I don't know by heart but it's possible that this changed in Annex60, yes.

**Comment by rubenbaetens** [comment]
I propose to merge `Annex60` back into `IDEAS` before v0.4.0 after https://github.com/iea-annex60/modelica-annex60/pull/412 which removes the `dynamicBalance` parameter.

**Comment by Mathadon** [comment]
will do, in due time

**Comment by rubenbaetens** [comment]
@damienpicard @Mathadon Mention the additional parameter in the construction-record.

**Comment by Mathadon** [comment]
Default IDEAS implementation is now with non-linear convection equations

**Comment by rubenbaetens** [comment]
Gentlemen, LBNL published their `Buildings 3.0.0` yesterday. aybe it's time we try to finish our 0.4 milestone ?

**Comment by Mathadon** [comment]
Yup, @damienpicard and @GlennReynders can you also verify if the changes in #460 are ok?

**Comment by damienpicard** [comment]
Here are the recurrent changes that I had to do to update my models to the new IDEAS version 0.3:

- Replace 
	- `IDEAS.SimInfoManager` by `IDEAS.BoundaryConditions.SimInfoManager`
	- `IDEAS.Interfaces.BaseClasses.Structure` by `IDEAS.Templates.Interfaces.BaseClasses.Structure`
	- `IDEAS.Interfaces.BaseClasses.HeatingSystem` by `IDEAS.Templates.Interfaces.BaseClasses.HeatingSystem`
	- `IDEAS.Interfaces.BaseClasses.VentilationSystem` by `IDEAS.Templates.Interfaces.BaseClasses.VentilationSystem`	
	- `IDEAS.Interfaces.BaseClasses.Occupant` by `IDEAS.Templates.Interfaces.BaseClasses.Occupant`
	- `IDEAS.Fluid.Domestic_Hot_Water.DHW_RealInput` by `IDEAS.Fluid.Taps.BalancedTap_m_flow`
	- `IDEAS.Constants.East` by `IDEAS.Types.Azimuth.E`
	- `IDEAS.Constants.North` by `IDEAS.Types.Azimuth.N`
	- `IDEAS.Constants.West` by `IDEAS.Types.Azimuth.W`
	- `IDEAS.Constants.South` by `IDEAS.Types.Azimuth.S`
	- `IDEAS.Constants.Wall` by `IDEAS.Types.Tilt.Wall`
	- `IDEAS.Constants.Ceiling` by `IDEAS.Types.Tilt.Ceiling`
	- `IDEAS.Constants.Floor` by `IDEAS.Types.Tilt.Floor`
	- `IDEAS.Constants.Other` by `IDEAS.Types.Tilt.Other`
	- `IDEAS.Fluid.Valves` by `IDEAS.Fluid.Actuators.Valves.Simplified`

- Remove
	- modifier 'mSenFac' of `zone` (`corrCV` is now used instead)
	- modifier 'm' from 'IDEAS.Fluid.Actuators.Valves.Simplified.Valves' and use 'tau' instead.

**Comment by damienpicard** [comment]
Here is a python code which automate these replacements:

```
"""
This file search in a whole folder and replace a string by another one.
"""
import glob
import os
from fnmatch import fnmatch

def listFilesInSubDir(root, pattern, printFiles = False):
    """
    Return all files containing the pattern in the root directory
    :param root:
    :param pattern:
    :param printFiles:
    :return:
    """
    filesList = []
    for path, subdirs, files in os.walk(root):
        for name in files:
            for p in pattern:
                if fnmatch(name, p):
                    filesList.append(os.path.join(path, name))
                    if printFiles:
                        print filesList[-1]
    return filesList

def containString(filesList, string):
    """
    Print all file names from  filesList with contains string
    :param filesList:
    :param string:
    :return:
    """
    for f in filesList:
        for s in string:
            if s in open(f).read():
                print f

def replaceLine(filesList, replacements):
    """
    Replace keys by arguments from replacements for all files in filesList
    :param filesList:
    :param replacements:
    :return:
    """
    for f in filesList:
        lines = []
        with open(f) as infile:
            for line in infile:
                for src, target in replacements.iteritems():
                    line = line.replace(src, target)
                lines.append(line)
        with open(f, 'w') as outfile:
            for line in lines:
                outfile.write(line)

if __name__ == '__main__':
    root = 'C:/Users/u0085306/Documents/work/modelica/SmartGeotherm'
    pattern = ["*.mo"]
    rep = {'hsPar' : 'heaSysPar','venPar' : 'venSysPar'}

    containString(listFilesInSubDir(root, pattern, printFiles = False),rep.keys())

    #listFilesInSubDir(root, pattern, printFiles = True)
    replaceLine(listFilesInSubDir(root, pattern, printFiles = False), rep)
```

**Comment by rubenbaetens** [comment]
The code defaults for linearisation are 'convection = non-linear' and 'radiation = linearised'

**Comment by rubenbaetens** [comment]
This overviews syas that we:
-  Moved IDEAS.HeatingSystems to IDEAS.Circuits
-  Moved IDEAS.VentilationSystems to IDEAS.Circuits
-  Moved IDEAS.Fluid.BaseCircuits to IDEAS.Circuits
But this seems outdated already, cause it seems to be in `IDEAS.Templates.[x]`

**Comment by Mathadon** [comment]
Something else for the update:
- comfort / internal gains models
- replaceable zone air model
- parameters for linearisation
- Annex60 update up to commit ...

**Comment by Mathadon** [comment]
occupancy model
zone air model
zone template
renamed ports in zone
general restructuring

**Comment by Mathadon** [comment]
see https://github.com/iea-annex60/modelica-annex60/pull/607

**Comment by Mathadon** [comment]
all `flowPort` variables have been renamed to `port_a` or `port_b`. 
`TStar` has been converted into `TRad`
conversion script was added
added example model of a terraced house
added twin house model and validation
updated icon
merged annex 60 up to commit fb3cb554055b0baa5aae1994fc5ab53074425542
added model for solar irradiation on window frame
changed implementation of building shade model
fixed bug in view factor implementation
added documentation
added thermostatic radiator valve model
removed insulation parameters: these are now part of constructiontype
harmonised implementation of Perez with annex 60
simplified defining new constructions
cleaned up implementation of bestest models
new specialised window types

**Comment by Mathadon** [comment]
I incorporated all these changes in the readme in #620

**Comment by Mathadon** [comment]
This has been fixed in #621
